### PR TITLE
2.7

### PR DESCRIPTION
**Pull request by macohen**
### Description
Mostly updated to be compatible with OpenSearch 2.7. It doesn't publish an artifact yet, but does use some of the standard build conventions as the rest of OpenSearch. Not sure this should be merged to main until we're comfortable and it won't break the regular build process in main.

### Issues Resolved
None exactly. Should start creating issues for more work.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,145 +1,171 @@
...
-// see https://github.com/opensearch-project/OpenSearch/blob/0ba0e7cc26060f964fcbf6ee45bae53b3a9941d0/buildSrc/src/main/java/org/opensearch/gradle/precommit/DependencyLicensesTask.java
-dependencyLicenses {
-  mapping from: /lucene-.*/, to: 'lucene'
-  mapping from: /asm-.*/, to: 'asm'
-  mapping from: /RankyMcRankFace.*/, to: 'lucene'
-  //mapping from: /Ranky.*/, to: 'lucene'
-  mapping from: /compiler.*/, to: 'lucene'
-}
-
-sourceSets {
```
Comment (hasnain2808) [inline build.gradle:90]: Hi @macohen 

I am trying to understand why have we replaced the sourceSets with [this] (https://github.com/opensearch-project/opensearch-learning-to-rank-base/pull/14/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R151)

<details>
<summary>The change</summary>
```
sourceSets {
  main {
    resources {
      srcDirs = ["config"]
      includes = ["**/*.yml"]
    }
  }
}
```
</details>


I am unable to run the plugin locally (both debug and normal mode) due to this change
It is unable to access the mapping file `"fstore-index-mapping.json"`

<details>
<summary>Exact eror</summary>
```
{
    "error": {
        "root_cause": [
            {
                "type": "illegal_state_exception",
                "reason": "failed to create ltr feature store index with resource [fstore-index-mapping.json]"
            }
        ],
        "type": "illegal_state_exception",
        "reason": "failed to create ltr feature store index with resource [fstore-index-mapping.json]",
        "caused_by": {
            "type": "null_pointer_exception",
            "reason": null
        }
    },
    "status": 500
}
```
</details>

Without this change I am able to create `.ltrstore`

I might have missed details, I am starting to get my hands dirty

Comment (hasnain2808) [reply build.gradle:90]: The PR was merged

**Comment by macohen** [comment]
@gsingers would you be able to take a pass at a review here? I'd like to merge this and then create a 2.11 branch from this one. Do you see value in 2.8, 2.9, 2.10 releases for this plugin? I'd rather just skip ahead to the latest and move forward from there, but would like your input.

File: .github/workflows/release-old.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
     - name: Grant execute permission for gradlew
       run: chmod +x gradlew
     - name: Build with Gradle
-      run: ./gradlew -Dtests.security.manager=false build
+      run: ./gradlew -Dskip.integtests=true -Djava.security.manager=allow -Dbuild.snapshot=false
```
Comment (macohen) [inline .github/workflows/release-old.yml:28]: skipping integ tests is temporary until we can build some more (might be able to remove). allowing the security manager and changes to the plugin-security.policy removed securitymanager violations.

Comment (macohen) [reply .github/workflows/release-old.yml:28]: TermStatQuery tests also pass now.

**Comment by gsingers** [comment]
@macohen I will take a look next week.

**Pull request closed without merging**